pkg/controller/jobframework/setup.go

```diff
@@ -96,8 +94,9 @@ func (m *integrationManager) setupControllers(mgr ctrl.Manager, log logr.Logger,
 				return nil
 			}
 		}
-		if err := noop.SetupWebhook(mgr, cb.JobType); err != nil {
-			return fmt.Errorf("%s: unable to create noop webhook: %w", fwkNamePrefix, err)
+		webhookFactory := NewSetupWebhookFactory(cb.JobType)
+		if err := webhookFactory(mgr, opts...); err != nil {
+			return fmt.Errorf("%s: unable to create default webhook: %w", fwkNamePrefix, err)
```

Comment:
```suggestion
		if err := SetupNoopWebhook(mgr, cb.JobType); err != nil {
			return fmt.Errorf("%s: unable to create noop webhook: %w", fwkNamePrefix, err)
```

The indirection is no longer needed.